compatibility-suite/pact-compatibility-suite/features/V3/matching_rules.feature

```diff
@@ -212,7 +212,7 @@ Feature: V3 era Matching Rules
       | application/octet-stream | file: sample.pdf |
     When the request is compared to the expected one
     Then the comparison should NOT be OK
-    And the mismatches will contain a mismatch with error "$" -> "Expected binary contents to have content type 'image/jpeg' but detected contents was 'application/pdf'"
```

Comment:
This change should be done in the compatibility suite repo, and it should only return one inferred type.

Reply:
cheers for the heads up, ill revert the mismatch message back to its original state and add debug logging in the matching method to state which detection method we have tried